src/groovy/org/transmart/FederatedUserDetailsService.groovy

```diff
@@ -111,6 +112,13 @@ public class FederatedUserDetailsService implements SAMLUserDetailsService {
 
             AuthUser newUser = AuthUser.createFederatedUser(federatedId,
                     username, realName, email, sessionFactory.currentSession);
+
+            if (attributeConfig.newUserAuthorities) {
+                // if new user authorities specified then replace default authorities
+                newUser.authorities.collect{}.each { newUser.removeFromAuthorities(it) }
```

Comment:
But you don't want to delete the authorities (`Role`s), you only want to dissociate the user from them, so the passage you cite is inapposite. The relevant from the page you link to is this one:

>    def a = Author.get(1)
>    a.books.clear()
>
> [...] [T]he books are still in the database. They are simply no longer associated with the author.